src/ripple/protocol/impl/RippleAddress.cpp

```diff
@@ -556,15 +573,30 @@ void RippleAddress::setAccountPrivate (
     setAccountPrivate (secretKey);
 }
 
-bool RippleAddress::accountPrivateSign (uint256 const& uHash, Blob& vucSig) const
+Blob RippleAddress::accountPrivateSign (Blob const& message) const
 {
-    vucSig = ECDSASign (uHash, getAccountPrivate());
-    const bool ok = !vucSig.empty();
+    if (vchData.size() == 33  &&  vchData[0] == 0xED)
+    {
+        uint8_t const*      secretKey = &vchData[1];
+        ed25519_public_key  publicKey;
+        Blob                signature (sizeof (ed25519_signature));
+
+        ed25519_publickey (secretKey, publicKey);
+
+        ed25519_sign (message.data(), message.size(), secretKey, publicKey, &signature[0]);
+
+        return signature;
+    }
+
+    uint256 const uHash = getSHA512Half (message);
+
+    Blob result = ECDSASign (uHash, getAccountPrivate());
+    bool const ok = !result.empty();
 
     CondLog (!ok, lsWARNING, RippleAddress)
             << "accountPrivateSign: Signing failed.";
 
-    return ok;
+    return result;
```

Comment:
I have no idea how signing could fail, either, but I presume it can, else we wouldn't check `ok` in the call to `CondLog`. I don't think we should throw because if a private key can trigger it, then it's possible user input could make it fail, and that's very bad.

And since I've noticed that at least one of the callers didn't check the flag in the old code anyway, callers could continue to ignore it with `std::tie`. However, I am strongly in favor of patterns like this that force the caller to check validity of results before using them, or to justify why they are not.